shapes.sc

```diff
@@ -373,17 +335,18 @@ tube_elipse(p1,p2,p3,material,width,rminor) ->(
     magProjMajor = __dot_prod(tv,hmajor);
     magProjMinor = __dot_prod(tv,hminor);
     
-    //this long if statement is the eqivlent of the check function from the python ver.
+    //These commented out boolan assignements show the various parts of the checks done in the volume call
     //inside=((magProjMajor^2)/(rmajor^2)+(magProjMinor^2)/(rminor^2))>=1;
     //outside=((magProjMajor)^2/(rmajor+width)^2+(magProjMinor)^2/(rminor+width)^2)<=1;
-    //print(str((magProjMajor)^2/(rmajor+width)^2)+'+'+str(magProjMinor));
-    //print(str(inside)+','+str(outside)+':'+str(tp));
-    started = (__dot_prod(tv,v1))>=0;
-    notended = (__dot_prod(tp-p2, -1*v1))>=0;
-    if( (((magProjMajor^2)/(rmajor^2)+(magProjMinor^2)/(rminor^2))>=1 &&
-          ((magProjMajor)^2/(rmajor+width)^2+(magProjMinor)^2/(rminor+width)^2)<=1 
-          && (__dot_prod(tv,v1))>=0 && (__dot_prod(tp-p2, -1*v1))>=0 ),
-      set(pos(_),material),
+    //
+    //
+    //started = (__dot_prod(tv,v1))>=0;
+    //notended = (__dot_prod(tp-p2, -1*v1))>=0;
+    // these are now arranged to run the most exclusive operation first that way once that fails it skips the other checks and continues
+    if( (((magProjMajor)^2/(rmajor+width)^2+(magProjMinor)^2/(rminor+width)^2)<=1 && 
+          ((magProjMajor^2)/(rmajor^2)+(magProjMinor^2)/(rminor^2))>=1 &&
+          (__dot_prod(tv,v1))>=0 && (__dot_prod(tp-p2, -1*v1))>=0 ),
+      set(pos(_),material), // I'd really like to add a feature to do replace more like the vanilla /fill command 

```

Comment:
On this: the more complex scarpet edit project kinda does that. I'd love to eventually integrate this app (and my other ones) into that one.